skimage/__init__.py

```diff
@@ -130,7 +130,17 @@ def _raise_build_error(e):
                              img_as_ubyte,
                              img_as_bool,
                              dtype_limits)
-    from .data import data_dir
     from .util.lookfor import lookfor
 
+    if sys.version_info < (3, 7):
+        from .data import data_dir
+
+def __getattr__(name):
```

Comment:
very nice use of py3.7+ lazy loading :-).

Reply:
Yeah, lets work on this in a bit. I want to get this PR to be really tiny, since it is hard to test in a pytest way.

Reply:
Honestly that was my quick and dirty attempt. I am convinced there is an elegant solution here without even global variables, but it's been a long week (gave a skimage+napari workshop today) and I'm just not seeing it. I'm happy to have another pass at this in a few days after recharging my batteries, but anyone should feel free to give it a go in the meantime!

Reply:
I think image_fetcher needs to be made a class.... Was thinking about this problem in the shower and you know that is where the best ideas come!

Reply:
You're right. The whole problem is that we have to keep track of a lot of state — that's what classes are for! I was stubbornly treating the whole module as a big class.